src/js_sync.hpp

```diff
@@ -724,26 +721,37 @@ void SyncClass<T>::set_sync_log_level(ContextType ctx, ObjectType this_object, A
 
     auto app = *get_internal<T, AppClass<T>>(ctx, Value::validated_to_object(ctx, args[0], "app"));
     std::string log_level = Value::validated_to_string(ctx, args[1], "log level");
-    std::istringstream in(log_level); // Throws
-    in.imbue(std::locale::classic()); // Throws
-    in.unsetf(std::ios_base::skipws);
-    util::Logger::Level log_level_2 = util::Logger::Level();
-    in >> log_level_2; // Throws
-    if (!in || !in.eof())
-        throw std::runtime_error("Bad log level");
-    app->sync_manager()->set_log_level(log_level_2);
+
+    common::Logger logger;
+    auto level = logger.get_level(log_level);
+    app->sync_manager()->set_log_level(level);
 }
 
-#if REALM_PLATFORM_NODE
 template<typename T>
 void SyncClass<T>::set_sync_logger(ContextType ctx, ObjectType this_object, Arguments &args, ReturnValue &return_value) {
     args.validate_count(2);
 
     auto app = *get_internal<T, AppClass<T>>(ctx, Value::validated_to_object(ctx, args[0], "app"));
     auto callback_fn = Value::validated_to_function(ctx, args[1], "logger_callback");
-    app->sync_manager()->set_logger_factory(*new realm::node::SyncLoggerFactory(ctx, callback_fn));
+    
+    Protected<typename T::GlobalContext> protected_ctx(Context<T>::get_global_context(ctx));
+    Protected<FunctionType> protected_callback(ctx, callback_fn);
+
+    auto show_logs = [=](auto level, auto message){
+        HANDLESCOPE(protected_ctx)
+
+        ValueType arguments[2] = {
+            Value::from_number(protected_ctx, level),
+            Value::from_string(protected_ctx, message)
+        };
+
+        Function::callback(protected_ctx, protected_callback, typename T::Object(), 2, arguments);
+    };
+
+    common::Logger logger;
+    auto sync_logger = logger.build_sync_logger(std::move(show_logs));
+    app->sync_manager()->set_logger_factory( *sync_logger );
 }
```

Comment:
[Code Quality]  I believe we agreed on documenting functions that we added or substantially refactored in comments.  I'd suggest at least documenting inputs, outputs, and purpose of the function.